src/app/modules/did-book/components/did-book-form/did-book-form.component.ts

```diff
@@ -25,7 +25,13 @@ export class DidBookFormComponent implements OnInit {
     if (!dids) {
       return;
     }
-    this.form.get('did').addValidators(ListValidator.stringExist(dids));
+    this.form
+      .get('did')
+      .setValidators([
+        Validators.required,
+        HexValidators.isDidValid(),
+        ListValidator.stringExist(dids),
+      ]);
```

Comment:
Was this the root cause of the error then? The validators weren't correctly being set? Or rather that something a missing or invalid DID was passed in?

Reply:
Validation was wrong. Steps to repro:
1. Add new DID contact
2. Remove this DID contact
3. try to add the same contact as in 1 (same DID). -> error show up, that this DID is already used. which is wrong obviously. The validators weren't cleaning up, this fixed the issue with cleaning up validators.

Reply:
But this fixes different issue. I hope that retryWhenWithDelay will fix it. If not, then i will try to find another solution (i couldn't reproduce it locally).